Pattern idea: Crossing the chasm (#368)
gruetter/innerSourceTest: patterns/1-initial/crossing-chasm.md

* Provide teams with a self check to scale the initiative (see [Maturity Model](../2-structured/maturity-model.md).

### Steps for reaching conservatives

* Include InnerSource training material in mandatory training courses for new
hires.
* Integrate any tooling and any experiments you have run into established
company structures (e.g. GitHub/ GitLab instances are under a regular plan,
accounts are easy to create and integrated with e.g. coporate single sign on
solutions, in-house support can provide help for these tools).
* Provide low effort opportunities for learning (e.g. put little riddles in your
coffee kitchen or whatever the digital alternative is for your company.)
* Ship bite sized learning pieces to inboxes.
* Pull in external expertise - sometimes in house expertise counts much less
than external statements.

### Steps for laggards

* Incorporate InnerSource into career competencies description.

## Resulting context

* The InnerSource initiative can reach teams with various innovation
preferences.
* Overall InnerSource adoption improves.
* Friction between teams decreases due to a common understanding.

## Known Instances

TBD

## Status

> Initial

Reviewer: ```suggestion
Initial

## References

- Book: [Crossing the Chasm](https://www.goodreads.com/book/show/61329.Crossing_the_Chasm) - Geoffrey A. Moore, Regis McKenna
- [Making InnerSource & Developer Experience Real at one of Canada's Top 5 Banks](https://youtu.be/QbJ4VFk8DwA?t=549) - GitHub Universe 2021 - Anthony Vacca 

```

The [InnerSource guidance group](add link here) should make different offers for
mastering InnerSource depending on where on the innovation scale employees are:

### Steps for reaching innovators

* Meet with people in person, in 1-on-1 settings, choose those in favour and
those against and those interested in the topic to create a first network.
* Start with InnerSource pilots.
* Create a community of practice that operates according to the principles that
you want to establish.
* Bring innovators together so they can support each other.
* Write down everything that you discuss in person or in calls so others that
lack the time to join those calls can follow along, provide their input and you
have a document that you can share with individuals following your trail.

Write down solutions to any problems you find along the way and share those
written solutions with people following your trail.

### Steps for reaching early adopters/ visionaries

* Share results and learnings form retrospectives of InnerSource pilots.
* Share external material (bonus points if that is material that colleagues
created earlier: talks, blog articles, book chapters, magazine articles, videos
etc.)

> * Co-create a document of shared [InnerSource principles](explicit-innersource-principles.md).

Reviewer: ```suggestion
* Co-create a document of shared [InnerSource Principles](../1-initial/explicit-innersource-principles.md).
```

### Steps for pragmatists

* Offer in-house and online training.
* Establish local experts on topics in the teams you want to reach.
* Provide an InnerSource strategy that is backed by higher level management.
* Provide InnerSource cheat sheet in particular for those new to the company.

> * Provide teams with a self check to scale the initiative (see [InnerSource
> maturity model](add link here).

Reviewer: ```suggestion
* Provide teams with a self check to scale the initiative (see [Maturity Model](../2-structured/maturity-model.md).
```

### Steps for reaching conservatives

* Include InnerSource training material in mandatory training courses for new
hires.
* Integrate any tooling and any experiments you have run into established
company structures (e.g. GitHub/ GitLab instances are under a regular plan,
accounts are easy to create and integrated with e.g. coporate single sign on
solutions, in-house support can provide help for these tools).
* Provide low effort opportunities for learning (e.g. put little riddles in your
coffee kitchen or whatever the digital alternative is for your company.)
* Ship bite sized learning pieces to inboxes.
* Pull in external expertise - sometimes in house expertise counts much less
than external statements.

> ### Steps for lagers

Reviewer 2: Is that supposed to be

```suggestion
### Steps for laggards
```

?

## Title

Crossing the InnerSource chasm

## Patlet

Early InnerSource experiments have been successful. Methods that were successful
convincing early teams stop working though when scaling the initiative. This
chasm can be crossed by using different methods to reach people at different
stages of the innovation curve.

## Problem

> Often InnerSource programs [start as experiments](add link here). From there

Reviewer: ```suggestion
Often InnerSource programs [start as experiments](../2-structured/start-as-experiment.md). From there
```

## Context

A corporation with teams on a wide spectrum of innovation friendliness has
stared an InnerSource initiative. Goals are set, early experiments were
successful, some teams are successfully using the new collaboration best
practices.

However there are teams which are less innovation friendly. Maybe they are more
risk averse than others. Maybe they prefer stable processes so they can focus on
their daily engineering challenges. Maybe they prefer to follow a proven path
instead of running into all of the hurdles that early adopters run into.

## Forces

- The goal is to increase collaboration, reduce duplication, increase knowledge
  sharing.
- Some teams already adopted a lot of InnerSource best practices, often when
  doing so they had to fix hurdles within the organisation when adopting a more
collaborative way of working.
- Some associates refuse to invest time in experimenting with new ways of
  working, preferring a stable environment.
- There is too much time to get everyone on board individually.
- Depending on where associates are on the innovation curve they are willing to
  invest different amounts of time to master a new way of working.

## Solution

> The [InnerSource guidance group](add link here) should make different offers for

Reviewer: ```suggestion
The [InnerSource guidance group](../1-initial/innersource-guidance-group.md) should make different offers for
```

@MaineC did you mean this pattern?

mastering InnerSource depending on where on the innovation scale employees are:

### Steps for reaching innovators

* Meet with people in person, in 1-on-1 settings, choose those in favour and
those against and those interested in the topic to create a first network.
* Start with InnerSource pilots.
* Create a community of practice that operates according to the principles that
you want to establish.
* Bring innovators together so they can support each other.
* Write down everything that you discuss in person or in calls so others that
lack the time to join those calls can follow along, provide their input and you
have a document that you can share with individuals following your trail.

Write down solutions to any problems you find along the way and share those
written solutions with people following your trail.

### Steps for reaching early adopters/ visionaries

* Share results and learnings form retrospectives of InnerSource pilots.
* Share external material (bonus points if that is material that colleagues
created earlier: talks, blog articles, book chapters, magazine articles, videos
etc.)

> * Co-createa document of shared [InnerSource principles](add link here).

Reviewer: ```suggestion
* Co-create a document of shared [InnerSource principles](explicit-innersource-principles.md).
```

* Provide teams with a self check to scale the initiative (see [InnerSource
maturity model](add link here).

### Steps for reaching conservatives

* Include InnerSource training material in mandatory training courses for new
hires.
* Integrate any tooling and any experiments you have run into established
company structures (e.g. GitHub/ GitLab instances are under a regular plan,
accounts are easy to create and integrated with e.g. coporate single sign on
solutions, in-house support can provide help for these tools).
* Provide low effort opportunities for learning (e.g. put little riddles in your
coffee kitchen or whatever the digital alternative is for your company.)
* Ship bite sized learning pieces to inboxes.
* Pull in external expertise - sometimes in house expertise counts much less
than external statements.

### Steps for lagers

* Incorporate InnerSource into career competencies description.

## Resulting context

* The InnerSource initiative can reach teams with various innovation
preferences.
* Overall InnerSource adoption improves.
* Friction between teams decreases due to a common understanding.

## Known Instances

TBD

## Status

Initial

> 

Reviewer: ```suggestion
```

  working, preferring a stable environment.
- There is too much time to get everyone on board individually.
- Depending on where associates are on the innovation curve they are willing to
  invest different amounts of time to master a new way of working.

## Solution

The [InnerSource guidance group](add link here) should make different offers for
mastering InnerSource depending on where on the innovation scale employees are:

### Steps for reaching innovators

* Meet with people in person, in 1-on-1 settings, choose those in favour and
those against and those interested in the topic to create a first network.
* Start with InnerSource pilots.
* Create a community of practice that operates according to the principles that
you want to establish.
* Bring innovators together so they can support each other.
* Write down everything that you discuss in person or in calls so others that
lack the time to join those calls can follow along, provide their input and you
have a document that you can share with individuals following your trail.

Write down solutions to any problems you find along the way and share those
written solutions with people following your trail.

> 

Reviewer: ```suggestion
```

* Provide teams with a self check to scale the initiative (see [InnerSource
maturity model](add link here).

### Steps for reaching conservatives

* Include InnerSource training material in mandatory training courses for new
hires.
* Integrate any tooling and any experiments you have run into established
company structures (e.g. GitHub/ GitLab instances are under a regular plan,
accounts are easy to create and integrated with e.g. coporate single sign on
solutions, in-house support can provide help for these tools).
* Provide low effort opportunities for learning (e.g. put little riddles in your
coffee kitchen or whatever the digital alternative is for your company.)
* Ship bite sized learning pieces to inboxes.
* Pull in external expertise - sometimes in house expertise counts much less
than external statements.

### Steps for lagers

* Incorporate InnerSource into career competencies description.

## Resulting context

* The InnerSource initiative can reach teams with various innovation
preferences.
* Overall InnerSource adoption improves.
* Friction between teams decreases due to a common understanding.

## Known Instances

TBD

## Status

Initial

> 

Reviewer: ```suggestion
```

* Include InnerSource training material in mandatory training courses for new
hires.
* Integrate any tooling and any experiments you have run into established
company structures (e.g. GitHub/ GitLab instances are under a regular plan,
accounts are easy to create and integrated with e.g. coporate single sign on
solutions, in-house support can provide help for these tools).
* Provide low effort opportunities for learning (e.g. put little riddles in your
coffee kitchen or whatever the digital alternative is for your company.)
* Ship bite sized learning pieces to inboxes.
* Pull in external expertise - sometimes in house expertise counts much less
than external statements.

### Steps for laggards

* Incorporate InnerSource into career competencies description.

## Resulting context

* The InnerSource initiative can reach teams with various innovation
preferences.
* Overall InnerSource adoption improves.
* Friction between teams decreases due to a common understanding.

## Known Instances

TBD

## Status

Initial

## References

- Book: [Crossing the Chasm](https://www.goodreads.com/book/show/61329.Crossing_the_Chasm) - Geoffrey A. Moore, Regis McKenna

> - [Making InnerSource & Developer Experience Real at one of Canada's Top 5 Banks](https://youtu.be/QbJ4VFk8DwA?t=549) - GitHub Universe 2021 - Anthony Vacca 

Reviewer: ```suggestion
- [Making InnerSource & Developer Experience Real at one of Canada's Top 5 Banks](https://youtu.be/QbJ4VFk8DwA?t=549) - GitHub Universe 2021 - Anthony Vacca
```
```suggestion
- [Making InnerSource & Developer Experience Real at one of Canada's Top 5 Banks](https://youtu.be/QbJ4VFk8DwA?t=549) - GitHub Universe 2021 - Anthony Vacca 
```
